app/src/main/java/com/example/caregiver/services/BeaconScanService.java

```diff
@@ -253,4 +252,69 @@ private String convertUnixToTimestamp(long unixTime) {
     private long getTimeNow() {
         return Calendar.getInstance().getTimeInMillis();
     }
+
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    private void createNotificationChannel() {
+        final NotificationManager notificationManager = (NotificationManager) getSystemService(
+                Context.NOTIFICATION_SERVICE);
+        if (notificationManager == null)
+            return;
+
+        final NotificationChannel channel = new NotificationChannel(DEFAULT_CHANNEL_ID, NOTIFICATION_CHANEL_NAME,
+                NotificationManager.IMPORTANCE_DEFAULT);
+        notificationManager.createNotificationChannel(channel);
+    }
+
+    private void sendTaskNotification(String roomName) {
+        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+        String userId = user.getUid();
+        final DatabaseReference database = FirebaseDatabase.getInstance().getReference();
+        final DatabaseReference rooms = database.child("/users/" + userId + "/rooms");
+
+        // Open TaskCaregivee page when user clicks on notification
+        Intent intent = new Intent(this, Dashboard.class);
+        intent.putExtra("fragment", "TaskCaregivee");
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, intent, 0);
+
+        // query tasks for the room the user is in
+        rooms.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot snapshot) {
+                for (DataSnapshot roomSnapShot : snapshot.getChildren()) {
+                    if (roomSnapShot.getKey().equals(roomName)) {
+                        for (DataSnapshot task : roomSnapShot.child("tasks").getChildren()) {
+                            // if the user has pending tasks in this room, send them notification
+                            String assignedStatus = (String) task.child("assignedStatus").getValue();
+                            if (assignedStatus.equals("true") || assignedStatus.equals("True")) {
+                                String contentText = String.format("You have tasks in the %s", roomName);
+                                // send notification
+                                createAndSendNotification(contentText, pendingIntent);
+                                return;
+                            }
+                        }
+                    }
+                }
+
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError error) {
+                Log.e("TaskNotification", "sendTaskNotifications failed", error.toException());
+            }
+        });
+    }
+
+    // sends notification to the user with specified "contentText"
+    private void createAndSendNotification(String contentText, PendingIntent pendingIntent) {
```

Comment:
Maybe we can change this function name? It's unclear what the difference is between sendTaskNottification() and createAndSendNotification()